pkg/feature-gates/feature-gates.go

```diff
@@ -20,9 +20,8 @@ type FeatureGates struct {
 }
```

Comment:
I probably should have suggested this in #1283 But it might make sense to make an interface with a single function in it:

```go
type FeatureGates interface {
  HonorWaitForFirstConsumerEnabled() (bool, error) 
}
```
And implement the rest as you have, the struct needs a slightly different name. The point of this is so that when we are writing unit tests, we can implement this interface in a test structure with a particular value and we don't need the runtime library client to make it work in the tests.

Then adding new feature gates will be a matter of adding functions on the interface.

Reply:
very good idea, I would like to create a new PR for that, no here.

Reply:
actually it might be easy to do and useful right now, in progress...

Reply:
done